code/modules/mod/modules/modules_timeline.dm

```diff
@@ -0,0 +1,336 @@
+	var/obj/structure/chrono_field/field = null
+	///where the chronofield maker was when the field went up
+	var/turf/startpos
+
+/obj/item/mod/module/tem/on_select_use(atom/target)
+	. = ..()
+	if(!.)
+		return
+	//trying to fire again, so disconnect what we have
+	if(field)
+		field_disconnect(field)
+	//fire projectile
+	var/obj/projectile/energy/chrono_beam/projectile = new /obj/projectile/energy/chrono_beam(get_turf(src))
+	playsound(src, 'sound/items/modsuits/time_anchor_set.ogg', 50, TRUE)
+	projectile.tem = src
+	projectile.firer = mod.wearer
+	projectile.fired_from = src
+	projectile.fire(get_angle(mod.wearer, target), target)
+
+/obj/item/mod/module/tem/on_uninstall()
+	if(field)
+		field_disconnect(field)
+
+/obj/item/mod/module/tem/proc/field_connect(obj/structure/chrono_field/F)
+	var/mob/living/user = loc
+	if(F.tem)
+		if(isliving(user) && F.captured)
+			to_chat(user, span_alert("<b>FAIL: <i>[F.captured]</i> already has an existing connection.</b>"))
+		field_disconnect(F)
+	else
+		startpos = get_turf(mod.wearer)
+		field = F
+		F.tem = src
+		if(isliving(user) && F.captured)
+			to_chat(user, span_notice("Connection established with target: <b>[F.captured]</b>"))
+
+/obj/item/mod/module/tem/proc/field_disconnect(obj/structure/chrono_field/F)
+	if(F && field == F)
+		var/mob/living/user = loc
+		if(F.tem == src)
+			F.tem = null
+		if(isliving(user) && F.captured)
+			to_chat(user, span_alert("Disconnected from target: <b>[F.captured]</b>"))
+	field = null
+	startpos = null
+
+/obj/item/mod/module/tem/proc/field_check(obj/structure/chrono_field/F)
+	if(F)
+		if(field == F)
+			var/turf/currentpos = get_turf(src)
+			var/mob/living/user = loc
+			if(currentpos == startpos && isliving(user) && user.body_position == STANDING_UP && !HAS_TRAIT(user, TRAIT_INCAPACITATED) && (field in view(CHRONO_BEAM_RANGE, currentpos)))
+				return TRUE
+		field_disconnect(F)
+		return FALSE
+
+/obj/projectile/energy/chrono_beam
+	name = "eradication beam"
+	icon_state = "chronobolt"
+	range = CHRONO_BEAM_RANGE
+	nodamage = TRUE
+	///reference to the tem... given by the tem!
+	var/obj/item/mod/module/tem/tem
+
+/obj/projectile/energy/chrono_beam/on_hit(atom/target)
+	if(target && tem && isliving(target))
+		var/obj/structure/chrono_field/F = new(target.loc, target, tem)
+		tem.field_connect(F)
+
+/obj/structure/chrono_field
+	name = "eradication field"
+	desc = "An aura of time-bluespace energy."
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "chronofield"
+	density = FALSE
+	anchored = TRUE
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF | FREEZE_PROOF
+	move_resist = INFINITY
+	interaction_flags_atom = NONE
+	var/mob/living/captured
+	var/obj/item/mod/module/tem/tem
+	var/timetokill = 30
+	var/mutable_appearance/mob_underlay
+	var/RPpos = null
+	var/attached = TRUE //if the tem arg isn't included initially, then the chronofield will work without one
```

Comment:
Again here, all of this need to be doc'd and references need to be handled (I know you're not touching this old code much, but might as well modernize it if you're planning on re-introducing it)